subsys/fs/fs.c

```diff
@@ -149,12 +149,16 @@ int fs_open(struct fs_file_t *zfp, const char *file_name, fs_mode_t flags)
 
 	zfp->mp = mp;
 
-	if (zfp->mp->fs->open != NULL) {
-		rc = zfp->mp->fs->open(zfp, file_name, flags);
-		if (rc < 0) {
-			LOG_ERR("file open error (%d)", rc);
-			return rc;
-		}
+#ifdef CONFIG_FILE_SYSTEM_NOTSUP_CHECK
```

Comment:
dev-review meeting:

We think it would make sense to use `CHECKIF` instead, because it's existing infrastructure.

```
CHECKIF(zfp->mp->fs->open == NULL) {
    return -ENOTSUP;
}
```

Reply:
Will use it.

Reply:
Currently I struggle to find C and C++ compatible solution for the macro part of the review, so I have decided to spun off the NULL-checking changes to another review:https://github.com/zephyrproject-rtos/zephyr/pull/29613; that one is non-RFC